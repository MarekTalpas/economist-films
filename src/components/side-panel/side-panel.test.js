// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import SidePanel from './side-panel';

const dummyUser = {id: 1, name: 'xxx', imgUrl: 'yyy'};

test('HomeContainer renders correctly', () => {
  const user = {id: 1, name: 'xxx', imgUrl: 'yyy'};
  const component : any = renderer.create(<SidePanel user={dummyUser} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('CheckboxWithLabel changes the text after click', () => {
  // Render a checkbox with label in the document
  const sidePanel = shallow(<SidePanel user={dummyUser} />);
  expect(sidePanel.hasClass('side-panel--expanded')).toEqual(false);
  sidePanel.find('.activator').simulate('click');
  expect(sidePanel.hasClass('side-panel--expanded')).toEqual(true);
});
