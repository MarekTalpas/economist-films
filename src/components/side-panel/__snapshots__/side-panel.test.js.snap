exports[`test HomeContainer renders correctly 1`] = `
<div
  className="side-panel ">
  <ul
    className="side-panel__option-list">
    <li
      className="side-panel__option">
      <button
        className="temp-icon activator"
        onClick={[Function]}>
        yyy1
      </button>
      <span
        className="side-panel__option-title">
        xxx
      </span>
    </li>
    <li
      className="side-panel__option side-panel__option--active">
      <button
        className="temp-icon" />
      <span
        className="side-panel__option-title">
        Search
      </span>
    </li>
    <li
      className="side-panel__option">
      <button
        className="temp-icon" />
      <span
        className="side-panel__option-title">
        Home
      </span>
    </li>
    <li
      className="side-panel__option">
      <button
        className="temp-icon" />
      <span
        className="side-panel__option-title">
        Favorites
      </span>
    </li>
  </ul>
</div>
`;
